monai/apps/pathology/datasets.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+from typing import Callable, List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+
+from monai.data import Dataset, SmartCacheDataset
+from monai.data.image_reader import WSIReader
+
+__all__ = ["PatchWSIDataset", "SmartCachePatchWSIDataset"]
+
+
+class PatchWSIDataset(Dataset):
+    """
+    This dataset read whole slide images, extract regions, and crate patches.
```

Comment:
could you run a spell checker for the docstrings...
```suggestion
    This dataset reads whole slide images, extracts regions, and creates patches.
```